ssl/ssl_test.go

```diff
@@ -0,0 +1,581 @@
+package ssl
+
+import (
+	"errors"
+	"strconv"
+	"testing"
+
+	"github.com/SUSE/scf-secret-generator/model"
+	"github.com/cloudflare/cfssl/csr"
+	cferr "github.com/cloudflare/cfssl/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	v1 "k8s.io/api/core/v1"
+)
+
+const CERT_ID = "cert-id"
+
+type MockBase struct {
+	mock.Mock
+}
+
+type MockLog struct {
+	MockBase
+}
+
+func (m *MockLog) Fatalf(str string, message ...interface{}) {
+	m.Called(str, message)
+}
+
+type MockSSL struct {
+	MockBase
+}
+
+func (m *MockSSL) createCA(secrets *v1.Secret, updates *v1.Secret, id string) bool {
+	args := m.Called(secrets, updates, id)
+	return args.Bool(0)
+}
+
+func (m *MockSSL) createCert(secrets *v1.Secret, updates *v1.Secret, id string) bool {
+	args := m.Called(secrets, updates, id)
+	return args.Bool(0)
+}
+
+func (m *MockSSL) updateCert(secrets *v1.Secret, updates *v1.Secret, id string) bool {
+	args := m.Called(secrets, updates, id)
+	return args.Bool(0)
+}
+
+func (m *MockBase) ClearCalls() {
+	m.Calls = nil
+	m.ExpectedCalls = nil
+}
+
+func TestRecordCertInfo(t *testing.T) {
+	assert := assert.New(t)
+
+	certInfo = make(map[string]CertInfo)
+
+	//
+	// Certificate is added to certInfo
+	//
+	configVar := model.ConfigurationVariable{
+		Name:   "CERT_NAME",
+		Secret: true,
+		Generator: &model.ConfigurationVariableGenerator{
+			ValueType: model.ValueTypeCertificate,
+			ID:        CERT_ID,
+		},
+	}
+	RecordCertInfo(&configVar)
+	assert.Equal(certInfo[CERT_ID].CertificateName, "cert-name")
+
+	//
+	// PrivateKey is added to certInfo
+	//
+	configVar = model.ConfigurationVariable{
+		Name:   "PRIVATE_KEY_NAME",
+		Secret: true,
+		Generator: &model.ConfigurationVariableGenerator{
+			ValueType: model.ValueTypePrivateKey,
+			ID:        CERT_ID,
+		},
+	}
+	RecordCertInfo(&configVar)
+	assert.Equal(certInfo[CERT_ID].PrivateKeyName, "private-key-name")
+
+	//
+	// If existing ID with certname, privatekey is added to the same CertInfo
+	//
+	assert.Equal(certInfo[CERT_ID].CertificateName, "cert-name")
```

Comment:
Would it be possible to assert that the private key corresponds to the certificate?

Reply:
Turns out you can! Fixed in 90adfdb

I tested with another valid, but different private key, and it does fail:
`Expected nil, but got: &errors.errorString{s:"tls: private key does not match public key"}`